packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -624,20 +443,32 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+    const matchConfig = this.match?.path?.bind(scope);
+
+    // Set prefix to '/' if none on match properties are defined.
+    const prefixPathMatch = areMatchPropertiesUndefined(this.match)
+      ? '/'
+      : matchConfig?.prefixPathMatch;
```

Comment:
There is some weird disconnect between what the docs say, and what the code does, that I don't understand.

In the docs, you claim that `path` is by default `HttpRoutePathMatch.startingWith('/')`. But the code doesn't actually do that. Instead, the code only uses that if the `match` properties are all `undefined`.

So, my question is this. If I specify the following Route:

```ts
router.addRoute('Route', {
  routeSpec: appmesh.RouteSpec.http({
    match: {
      method: appmesh.HttpRouteMethod.GET,
  }),
});
```

What will be the _actual_ service behavior here, as far as path matching is concerned?

Reply:
_actual_ service would set `prefix` to `/`.

Reply:
Changed the logic to actually assign `prefix` with `/` when none of the path match are defined.